### ISSUE TITLE
CALLBACK_RESULT_URL_OK not fired on Android

### ISSUE THREAD

**Issue by dapetcu21** [defold/extension-webview#14]
I tried this with both `webview.open_raw()` and `webview.open()` (with a `data:` URL) on the following devices:

One Plus One (A0001), Android 6 (not sure exactly what point version, as I don't have access to it now)
Alcatel Idol 4 6055P, Android 6.0.1

Eval callbacks like `CALLBACK_RESULT_EVAL_OK` fire correctly. It's just `CALLBACK_RESULT_URL_OK` that doesn't work.

**Comment by dapetcu21** [comment]
Here's my code, for reference:

```lua
local sys_config = require "lib.sys_config"
local url_utils = require "lib.url_utils"

return function ()
  -- luacheck: globals webview
  if webview then
    local html_data = sys.load_resource("/resources/custom/licenses.html")
    local update_timer, close_request

    local shown = false
    local function show_webview(webview_id)
      if shown then return end
      shown = true

      webview.set_visible(webview_id, 1)
      update_timer = timer.delay(0.0, true, function ()
        if not close_request then
          close_request = webview.eval(webview_id, "window.shouldClose")
        end
      end)
    end

    local webview_id = webview.create(function (_, webview_id, request_id, type, data)
      if type == webview.CALLBACK_RESULT_URL_OK then
        show_webview(webview_id)

      elseif type == webview.CALLBACK_RESULT_URL_ERROR then
        print("ERROR in webview: " .. data.result)

      elseif type == webview.CALLBACK_RESULT_EVAL_OK then
        if request_id == close_request then
          close_request = nil
          if data.result == "true" then
            timer.cancel(update_timer)
            update_timer = nil
            webview.destroy(webview_id)
          end
        end

      end
    end)
    -- Workaround for open_raw not working on iOS
    if sys_config.system_name == "iPhone OS" then
      local data_url = "data:text/html," .. url_utils.url_encode_component(html_data)
      webview.open(webview_id, data_url, { hidden = true })
    else
      webview.open_raw(webview_id, html_data, { hidden = true })
    end

    -- Workaround for CALLBACK_RESULT_URL_OK not firing on Android
    if sys_config.system_name == "Android" then
      show_webview(webview_id)
    end

    return
  end

  local path = sys_config.bundle_root_path .. sys_config.path_sep .. "licenses.html"
  if sys_config.system_name == "Darwin" then
    path = sys_config.bundle_root_path .. "/Contents/Resources/licenses.html"
  end
  if sys_config.path_sep == "\\" then
    path = "/" .. (path:gsub("\\", "/"))
  end
  local url = "file://" .. url_utils.url_encode(path)
  print("Opening " .. url)
  sys.open_url(url)
end
```

**Comment by britzl** [comment]
PR incoming. Please check: https://github.com/defold/extension-webview/archive/Issue-14-android-callback-result-ok-not-fired.zip

**Comment by dapetcu21** [comment]
I just tried this. It didn't seem to do the trick. Didn't check exactly what happened, but I removed the workaround from the above code and no webview became visible.

**Comment by britzl** [comment]
iOS will be solved by #17 
I likely only fixed this for webview.open() on Android. I will look into webview.open_raw() as well.
